esx_service/utils/auth_api.py

```diff
@@ -241,7 +247,12 @@ def get_default_datastore(name):
 
 def _tenant_create(name, description="", vm_list=None, privileges=None):
     """ API to create a tenant """
+    global valid_tenant_name_reg
     logging.debug("_tenant_create: name=%s description=%s vm_list=%s privileges=%s", name, description, vm_list, privileges)
+    if not valid_tenant_name_reg.match(name):
```

Comment:
1. need to add testing done for this code snip
2. can you confirm about having tests exercising this new change?